apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -0,0 +1,745 @@
+        }
+    }
+
+    /**
+    * @dev Internal function to cast a vote. It assumes the queried vote exists.
+    */
+    function _vote(uint256 _voteId, bool _supports, address _voter) internal {
+        // This could re-enter, though we can assume the governance token is not malicious
+        Vote storage vote_ = votes[_voteId];
+        _castVote(vote_, _voteId, _supports, _voter);
+        _overwriteCasterIfNecessary(vote_, _voter);
+
+        if (_canExecute(vote_)) {
+            // We've already checked if the vote can be executed with `_canExecute()`
+            _unsafeExecuteVote(_voteId);
+        }
+    }
+
+    /**
+    * @dev Internal function for a representative to cast a vote on behalf of many voters. It assumes the queried vote exists.
+    */
+    function _voteOnBehalfOf(address[] _voters, uint256 _voteId, bool _supports) internal {
+        Vote storage vote_ = votes[_voteId];
+        for (uint256 i = 0; i < _voters.length; i++) {
+            address voter = _voters[i];
+            require(_canVote(vote_, voter), ERROR_CAN_NOT_VOTE);
+            require(!_withinOverruleWindow(vote_), ERROR_WITHIN_OVERRULE_WINDOW);
+            require(_isRepresentativeOf(voter, msg.sender), ERROR_NOT_REPRESENTATIVE);
+
+            if (_hasNotVotedYet(vote_, voter)) {
+                _voteOnBehalfOf(_voteId, _supports, voter, msg.sender);
+            } else {
+                emit ProxyVoteFailure(_voteId, voter, msg.sender);
+            }
+        }
+    }
+
+    /**
+    * @dev Internal function for a representative to cast a vote on behalf of a voter. It assumes the queried vote exists.
+    */
+    function _voteOnBehalfOf(uint256 _voteId, bool _supports, address _voter, address _representative) internal {
+        Vote storage vote_ = votes[_voteId];
+        _castVote(vote_, _voteId, _supports, _voter);
+        vote_.casters[_voter] = _representative;
+
+        emit ProxyVoteSuccess(_voteId, _voter, _representative, _supports);
+    }
+
+    /**
+    * @dev Internal function to execute a vote. It assumes the queried vote exists.
+    */
+    function _executeVote(uint256 _voteId) internal {
+        require(_canExecute(votes[_voteId]), ERROR_CAN_NOT_EXECUTE);
+        _unsafeExecuteVote(_voteId);
+    }
+
+    /**
+    * @dev Internal unsafe version of _executeVote that assumes you have already checked if the vote can be executed and exists
+    */
+    function _unsafeExecuteVote(uint256 _voteId) internal {
+        Vote storage vote_ = votes[_voteId];
+
+        vote_.executed = true;
+        vote_.disputableStatus = DisputableStatus.Closed;
+        _closeAgreementAction(vote_.actionId);
+
+        bytes memory input = new bytes(0); // TODO: Consider input for voting scripts
+        address[] memory blacklist = new address[](1);
+        blacklist[0] = address(_getAgreement());
+        runScript(vote_.executionScript, input, blacklist);
+        emit ExecuteVote(_voteId);
+    }
+
+    /**
```

Comment:
More of a meta note, but I would recommend putting one of those big headers for these disputable actions (perhaps even before the internal functions), to better group them.